scripts/build_binary.sh

```diff
@@ -30,7 +30,9 @@ mkdir -p $1
 
 cd "${ROOT}"
 
+version_ldflags="-X github.com/awslabs/amazon-ecr-credential-helper/ecr-login/version.VERSION=${2}"
```

Comment:
The behavior that you have for `version.VERSION` defaulting to `"master"` isn't quite working because you're still setting `version_ldflags` when `${2}` is empty.  Can you do this instead?

```bash
version_ldflags=""
if [[ -n "${2}" ]]; then
  version_ldflags="-X github.com/awslabs/amazon-ecr-credential-helper/ecr-login/version.VERSION=${2}"
fi
```

Reply:
Just realised that version will always be set, since the version is grabbed from the VERSION file. Do you want a new target for building it locally without the Version?

Also depends how much you want to follow semver, since semver supports snapshots or alplha versions between versions

Reply:
In this particular case (and why I noticed it), the `$(LOCAL_BINARY)` target isn't currently grabbing it from the `VERSION` file.

It should typically be set if you're building with `make`, but could be unset if it's build some other way.  I think having a default value is reasonable, but it's not necessary to have a target specifically for building without the version.

> Also depends how much you want to follow semver, since semver supports snapshots or alplha versions between versions

Yeah, using a prerelease tag (like `0.0.1-dev`) is a reasonable application of semver, but I don't think we need programmatic infrastructure around that yet.

Reply:
Fixed by 388652f